core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1794,20 +1811,30 @@ class Log(@volatile private var _dir: File,
     def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]) = {
       if (diff - segment.size >= 0) {
         diff -= segment.size
+        info(s"Segment with base offset ${segment.baseOffset} will be deleted due to" +
+          s" ${config.retentionSize} breach. Segment size is ${segment.size} and total log size" +
```

Comment:
nit: ".. will be deleted due to **retention size in bytes** ${config.retentionSize} breach."